osu.Game/Skinning/ArgonSkinTransformer.cs

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Game.Screens.Play.HUD;
+using osuTK;
+
+namespace osu.Game.Skinning
+{
+    public class ArgonSkinTransformer : SkinTransformer
+    {
+        public ArgonSkinTransformer(ISkin skin)
+            : base(skin)
+        {
+        }
+
+        public override Drawable? GetDrawableComponent(ISkinComponentLookup lookup)
+        {
+            if (lookup is SkinComponentsContainerLookup containerLookup
+                && containerLookup.Target == SkinComponentsContainerLookup.TargetArea.MainHUDComponents
+                && containerLookup.Ruleset != null)
+            {
+                return base.GetDrawableComponent(lookup) ?? new Container
```

Comment:
I've hopefully resolved conflicts, but I still have a lot of questions.

Will start with a simple question: why is this here now? Why does `ArgonSkinTransformer` have to exist? `Transformer`s were supposed to allow rulesets to do things, but you have changed what transformer means now, as this is being instantiated by `Ruleset`.

Reply:
On further reading and understanding of this PR, the probable answer is "because I wanted to be able to provide a per-skin default for cases where a ruleset doesn't do anything special", does that sound right?

Reply:
@frenzibyte I've applied 60d383448f22f36e9cc6c95f568cd81bb0ca5bef which removes this stuff. please give it 5 minutes and let me know if it makes sense within that time limit.

Reply:
That is correct.

Reply:
> @frenzibyte I've applied [60d3834](https://github.com/ppy/osu/commit/60d383448f22f36e9cc6c95f568cd81bb0ca5bef) which removes this stuff. please give it 5 minutes and let me know if it makes sense within that time limit.

I...don't know. It looks correct, but I feel it's introducing another level of complexity to the lookup system in my head / when parsing this mentally. I wouldn't say that the way I had it is better, though.

This is all cheap talk and I cannot easily imagine how it can be put to code, but, it definitely feels like there should be two methods:
 1. `Skin.GetDrawableComponent`, which tries to look up user configuration, then fall back to...
 2. `Skin.GetDefaultDrawableComponent`, which returns default skin implementation, and that can be transformed by a skin transformer somehow.

I'm proposing this for `Skin` specifically because it is the class that introduces the concept of "user configuration".

Reply:
> This is all cheap talk and I cannot easily imagine how it can be put to code, but, it definitely feels like there should be two methods

Yep, in previous discussions with smoogi this was unanimously agreed upon, but I'm not doing that here. It will be a separate step after all your changes are merged.

Reply:
As long as it's all temporary until a split happens then I'm fine looking away, I can't easily explain why this feels complicated in my head.